### ISSUE TITLE
won't open URL in Chrome

### ISSUE THREAD

**Issue by JohnGEyles** [MacPass/MacPass#742]
Just installed Macpass on mid-2011 Macbook Air running High Sierra.    "Double-click on URL -> Opens URL" is enabled in Prefs.   But when I double-click on URL, Chrome opens a new window, but it's just www.google.com, not the URL specified in the entry.  Some issue with my Chrome configuration ?

**Comment by JohnGEyles** [comment]
I have Macpass workflow prefs set exactly as you show.
I had Chrome set to "open the new tab page" but changed it to "continue where you left off" and it didn't help (although I haven't re-started Chrome).

I installed Macpass on my 2009 MacPro running 10.11 (El Cap) and it works a lot better there.    Opens the URL, but doesn't always do the autotype correctly; I attribute that more to spasticity at the website signin page though.

So it seems it is a problem with 10.13.

**Comment by JohnGEyles** [comment]
A related oddity (tell me if you want me to file this as another issue) ...

One of the signin URLs I have has a '#' character in it.    When Macpass copies the URL to the Chrome tab it opens, it replaces the '#' with "%23" (I believe 23 is hex for #).   

Maybe Macpass has some reason to need to handle '#' especially.   It's silly for that to be in the signin page's URL anyhow.

**Comment by JohnGEyles** [comment]
FWIW, KeepassXC is exhibiting pretty much identical behavior.   On 10.13 Macbook Air: doesn't open a double-clicked URL at all.   On 10.11 Mac Pro: opens the double-clicked URL but doesn't do the autotype.

**Comment by mstarke** [comment]
Thanks for the detailed information. I'll try to correlate this with any bugs filed to Apple or see if the docs for 10.13 include anything about changes to the NSWorkspace APIs (which are the ones used to open the URL)

**Comment by JohnGEyles** [comment]
Any hints why autotype is not working for me under 10.11 ?    Thanks for checking all this out.

**Comment by mstarke** [comment]
There are 3 issues in this one.

Could you please split the one with "#" into a separate one.
Autotype not working properly is also worth a ticket. This might be due to some timing issues since copy/paste timing might have changed. 10.10 required some changes there as well so this might be the same line of error but I have no absolute clue what's going on. The current master now uses a newer NSWorkspace API to open the URL. If the current continuous is up to date, can you maybe test this on your end to see if the initial Chrome related  error is at least fixed?

**Comment by sfiera** [comment]
The Chrome error sounds like https://crbug.com/777863 / rdar://35505509

**Comment by mstarke** [comment]
@sitsofe nice find! This seems to be the problem and the end result is that a radar is filed. I hope this get's fixed soon so I'll have no way on MacPass' side to prevent this behaviour from happening.

**Comment by sitsofe** [comment]
@mstarke I think the credit for this has to go to @sfiera rather than me :-)

**Comment by mstarke** [comment]
Darn. You're right. @sfiera the hero :)

**Comment by JohnGEyles** [comment]
I started a new issue for the '# in a URL' (and deleted the comment above).

I couldn't quite follow the linked thread on the 10.13 issue; do we just have to wait for the 'radar' (whatever that is) to be dealt with ?    Is restarting Chrome every time there's an auto-update a temporary fix ?

**Comment by sfiera** [comment]
I think that, according to that bug, there is a fix in Chrome v64. So:
* If you’re using Chrome Stable (v63; you probably are), then you should be fine once v64 reaches stable (2-4 weeks from now, if I recall).
* Until then, restarting Chrome for updates should fix the problem.
* If you’re using a prerelease version of Chrome (Beta, Dev, Canary), you should already have the fix.
* The fix is a workaround for an Apple bug (that’s what “radar” bugs are). That will be fixed by OS X 10.13.4 at the earliest, but that shouldn’t matter for you; Chrome should work fine before then.

**Comment by JohnGEyles** [comment]
Yes, I was using Chrome v63, restarted and it auto-updated to v64 (stable release channel was already  up to v64) and problem is solved.   Thanks.

My complaint about autotype was from a misapprehension about how it's supposed to work.   I thought I could double-click the URL and it'd open it and then type in the userid and password automatically (without my doing any action after double-clicking the URL).    But why couldn't it work this way ?

**Comment by JohnGEyles** [comment]
So now I'm getting this faulty behavior again: double-clicking the URL just opens an empty window.   I re-started Chrome as recently as yesterday, and have been running v64 since it seemed like this problem was resolved (previous comment, 8 days ago).    But when I check the version number, it shows v64 but also says "Nearly up to date! Relaunch Google Chrome to finish updating."    Is it possible that even though Macpass was working with my currently-installed version of Chrome, it now doesn't work because there's apparently a newer version ?

**Comment by sfiera** [comment]
I guess. I thought that the bug said that there was a workaround, and that the timing was right for v64. I must be wrong about one or both of those. In that case, I’m not sure what the timeline for a fix is.
